doc/OnlineDocs/index.rst

```diff
@@ -26,6 +26,7 @@ with a diverse set of optimization capabilities.
    library_reference/index.rst
    contribution_guide.rst
    contributed_packages/index.rst
+   related_packages.rst
```

Comment:
Style question: What about integrating this into the current "contrib" page?  I would like to rename that section of the documentation (and have been working on a slight reorganization over on #2006 see [here](https://pyomo-2006.org.readthedocs.build/en/2006/contributed_packages/index.html).  I think I want to get away from the "third party" moniker, but am struggling to come up with a good way to concisely describe the sliding scale between core, extensions, contrib, and external packages.

Reply:
Fine with me.

Reply:
I was thinking about this, and it might make sense to combine "Modeling Extensions", "Third Party Contributions", and "Related Packages" into a single "Extensions" page with sub-categories "Stable", "Beta", and "External". Thoughts?

Reply:
I like this idea, although we might want to consider different names for "Stable" and "Beta".  It is true that we are promising a higher lever of support and stability (at least for some features) for extensions outside of `contrib`, there is still active development in those packages (I fear "stable" would be read as "stagnant").  Similarly, I think there is some functionality in `contrib` that, while still evolving, is well out of what I would consider "beta".

Reply:
Good points...